### PR TITLE
Test for DiscontinuousTestODE

### PR DESCRIPTION
**Pull request by lisawim**
Since hardcoded solutions do not say anything about the correctness of the problem class itself, I also added a small test for the `DiscontinuousTestODE`. It's only a small one whether the correct event time together with value of state function at event is tested. In the TIME-X test hackathon I also learnt that errors can be tested by capture them. Thus, `ProblemError` and `logger.warning` are also captured through this test when `nan` values arises.

The function `get_switching_info` is not tested here, because this is rather stuff of `SwitchEstimator` which I also started to test during the test hackathon. (Coming soon some time..)

Let me know if this is not enough.

### CODE REVIEW

File: pySDC/tests/test_problems/test_DiscontinuousTestODE.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    args = {
+        'rhs': u0,
+        'dt': dt,
+        'u0': u0,
+        't': t0,
+    }
+
+    sol = DODE.solve_system(**args)
+    assert np.isclose(
+        sol[0], DODE.u_exact(t0 + dt)[0], atol=2e-3
```
[the remark this answers is not in the document]
Comment (lisawim) [reply pySDC/tests/test_problems/test_DiscontinuousTestODE.py:36]: Your are right, but in this case, the solution consists of only one value.

File: pySDC/tests/test_problems/test_DiscontinuousTestODE.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    problem_params = {
+        'newton_tol': 1e-13,
+    }
+
+    DODE = DiscontinuousTestODE(**problem_params)
+
+    u_event = DODE.u_exact(DODE.t_switch_exact)
+    h = u_event[0] - 5
+
+    assert h < 1e-15, 'Value of state function at exact event time is not zero!'
```
[the remark this answers is not in the document]
Comment (lisawim) [reply pySDC/tests/test_problems/test_DiscontinuousTestODE.py]: Done.

**Comment by pancetta** [comment]
Ready to be merged?

**Comment by lisawim** [comment]
From my side, yes. Depends on @brownbaerchen

**Pull request merged**